core/embed/bootloader/messages.c

```diff
@@ -44,6 +44,10 @@
 #include "emulator.h"
 #endif
 
+#if defined(TREZOR_MODEL_R) && !defined(TREZOR_EMULATOR)
```

Comment:
why is emulator excluded here? it is supposed to work and recognize whether it is unlocked

Reply:
I assumed that the emulator should always return true, but I hadn't realized that it sets up the memory to indicate that there is a secret stored in the corresponding sector. However, the firmware emulator doesn't seem to be doing that.

Reply:
Right, that might be a missing piece. I think I intended to add a cmdline option to the bootloader-emulator that would write a secret header.

For firmware emulator, i would leave it to always-true for now, but it seems worth making it work for the bootloader emulator.

Reply:
Correction, neither of them sets up the memory to indicate that there is a secret stored in the corresponding sector. So both will always return false and won't firmware allow authenticity checks if we use secret.c in the emulator. Of course we can include secret.h without using it, if that's what you mean in this comment.

Reply:
55238596fa265acc17e68d573bd4ac37d1cee290